sdv/constraints/tabular.py

```diff
@@ -211,13 +226,31 @@ def _fit(self, table_data):
             table_data (pandas.DataFrame):
                 The Table data.
         """
-        self._dtype = table_data[self._high].dtype
-        self._diff_column = f'#{self._low}#{self._high}'
+        if self._high_is_scalar is None:
+            self._high_is_scalar = self._high not in table_data.columns
+        if self._low_is_scalar is None:
+            self._low_is_scalar = self._low not in table_data.columns
+
+        if self._low_is_scalar:
+            self.constraint_columns = tuple(self._high)
+            self._dtype = table_data[self._high].dtype
+
+        if self._high_is_scalar:
+            self.constraint_columns = tuple(self._low)
+            self._dtype = table_data[self._low].dtype
+
+        if not self._low_is_scalar and not self._high_is_scalar:
+            self._dtype = table_data[self._high].dtype
+
         separator = '#'
-        while not self._valid_separator(table_data, separator, self.constraint_columns):
-            separator += '#'
+        self._diff_column = separator + separator.join(self.constraint_columns)
```

Comment:
@csala The reason that this doesn't use the `_valid_separator` method is that the `self.column_constraints` tuple might now only have one element. In this case, the `_valid_separator` method gets stuck in an infinite loop, because there is nothing to join the only column to, so it finds the column name inside the table. Not sure what you think is a good approach. I didn't use the `uuid` either because it will be good to know what the column name ends up being for integration tests. Also not sure how to keep the `uuid` from being something in the column names of the table

Reply:
I don't understand this. If we only validate against the columns not in the `self.constraint_columns`, it is possible to end up with a column name that is already present. If the `self.constraint_columns` contains `('a',)`, then the diff column should be called `'a#'`. But if we just join the columns it will return `'a'`